stefaniaPopescu26/assignment3/Dictionary.java

```diff
@@ -36,21 +58,70 @@ public void readData(String filename) {
         }
     }
 
-    public boolean isWord(String word) {
-        return containsKey(word);
+    private void add(String word) {
+        if (word.isEmpty()) {
+            return;
+        }
+
+        TrieNode node = root;
+
+         do {
+            char letter = word.charAt(0);
+
+            if (!node.children.containsKey(letter)) {
+                node.children.put(letter, new TrieNode());
+            }
+
+            node = node.children.get(letter);
+            word = word.substring(1);
```

Comment:
Here you are making a lot of heavy operations on strings -- substring creates a copy of your original string and then assigns it to the word variable. Iterating through characters in the original string will be much more efficient.